app/code/local/Dwyera/Pinpay/Model/PaymentMethod.php

```diff
@@ -38,16 +39,63 @@ public function assignData($data)
             $data = new Varien_Object($data);
         }
 
+        //Check if customer tokenization is enabled.
+        $block = new Dwyera_Pinpay_Block_Form;
+        $customerTokenizationEnabled = $block->iscustomerTokenizationEnabled();
+        $customerToken = $data->getCustomerToken();
+
+        //To Check if the customer is logged in.
+        $customerData = $this->getCustomer();
+
+        //Get the card Token from data
         $cardToken = $data->getCardToken();
+
+        //Check if customer token is enabled, customer is logged in and the card token is not the same as the one in database.
+        /* Comparing the card token with existing token to make sure the following things donot happen
+        1. The unnecessary API calls to the Pinpayment API is avoided if the card token is same
+         */
+        if($customerTokenizationEnabled && $customerData != false && $customerData->getData('pinpayment_card_token') != $cardToken){
```

Comment:
@ramya-aligent  in future instead of `$customerData != false` simply use `$customerData`
its more readable